lib/modem_info/modem_info.c

```diff
@@ -718,6 +737,132 @@ int modem_info_rsrp_register(rsrp_cb_t cb)
 	return 0;
 }
 
+/* FW UUID is 36 characters: XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX */
+#define FW_UUID_SIZE 37
+
+int modem_info_get_fw_uuid(char *buf, size_t buf_size)
+{
+	int ret;
+
+	if (buf == NULL || buf_size < FW_UUID_SIZE) {
+		return -EINVAL;
+	}
+
+	ret = nrf_modem_at_scanf("AT%XMODEMUUID",
+				 "%%XMODEMUUID: %" STRINGIFY(FW_UUID_SIZE) "[^\r\n]",
+				 buf);
+	if (ret != 1) {
+		LOG_ERR("Could not get FW ID, error: %d", ret);
+		return map_nrf_modem_at_scanf_error(ret);
+	}
+	return 0;
+}
+
+#define SWVER_FMT_STR_SIZE 23
```

Comment:
Add a documentation for this too like you did here -> https://github.com/nrfconnect/sdk-nrf/pull/8305/files#diff-e39ead1e68663cfe735958ac8126660743041b5a8bc65f519d5b173a206d6d3fR740